hi-ml-histopathology/src/histopathology/utils/report_utils.py

```diff
@@ -0,0 +1,339 @@
+#  -------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  -------------------------------------------------------------------------------------------
+
+import pickle
+from pathlib import Path
+from typing import Dict, Optional, Sequence
+
+import dateutil.parser
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from azureml.core import Experiment, Run, Workspace
+from matplotlib.axes import Axes
+from matplotlib.figure import Figure
+from matplotlib.lines import Line2D
+from sklearn.metrics import auc, precision_recall_curve, roc_curve
+
+from health_azure.utils import aggregate_hyperdrive_metrics, get_aml_run_from_run_id, get_tags_from_hyperdrive_run
+from histopathology.utils.naming import ResultsKey
+
+TRAIN_STYLE = dict(ls='-')
+VAL_STYLE = dict(ls='--')
+BEST_EPOCH_LINE_STYLE = dict(ls=':', lw=1)
+BEST_TRAIN_MARKER_STYLE = dict(marker='o', markeredgecolor='w', markersize=6)
+BEST_VAL_MARKER_STYLE = dict(marker='*', markeredgecolor='w', markersize=11)
+
+
+def download_from_run_if_necessary(run: Run, remote_dir: Path, download_dir: Path, filename: str) -> Path:
```

Comment:
Should this live in health_azure/utils? 

Also, could call the existing function `_download_file_from_run`, so if AML change the behaviour of `run.download_file`, we only have to update 1 place